packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/NotebookCellScopeHandler.ts

```diff
@@ -1,132 +1,70 @@
 import {
-  Range,
   type Direction,
-  type NotebookCell,
   type Position,
   type ScopeType,
   type TextEditor,
 } from "@cursorless/common";
 import type { LanguageDefinitions } from "../../../languages/LanguageDefinitions";
-import { ide } from "../../../singletons/ide.singleton";
-import { NotebookCellTarget } from "../../targets";
+import { BaseScopeHandler } from "./BaseScopeHandler";
+import { NotebookCellApiScopeHandler } from "./NotebookCellApiScopeHandler";
+import { OneOfScopeHandler } from "./OneOfScopeHandler";
 import type { TargetScope } from "./scope.types";
 import type {
+  ComplexScopeType,
   ScopeHandler,
   ScopeIteratorRequirements,
 } from "./scopeHandler.types";
+import type { ScopeHandlerFactory } from "./ScopeHandlerFactory";
 
-export class NotebookCellScopeHandler implements ScopeHandler {
+export class NotebookCellScopeHandler extends BaseScopeHandler {
   public readonly scopeType = { type: "notebookCell" } as const;
-  public readonly iterationScopeType = { type: "document" } as const;
-  public readonly includeAdjacentInEvery = false;
+  protected isHierarchical = false;
+  private readonly scopeHandler: ScopeHandler;
 
-  constructor(
-    private languageDefinitions: LanguageDefinitions,
-    _scopeType: ScopeType,
-    private languageId: string,
-  ) {}
-
-  *generateScopes(
-    editor: TextEditor,
-    position: Position,
-    direction: Direction,
-    hints: ScopeIteratorRequirements,
```

Comment:
NTS: The actual type of the interface is `requirements?: Partial<ScopeIteratorRequirements>`. This should really be a tsc compile error but it wasn't. We therefore started to read properties on `hints` when some of those properties could be undefined which threw an exception

Reply:
We needed something like what the base class does:

```
  *generateScopes(
    editor: TextEditor,
    position: Position,
    direction: Direction,
    requirements: Partial<ScopeIteratorRequirements> = {},
  ): Iterable<TargetScope> {
    const hints: ScopeIteratorRequirements = {
      ...DEFAULT_REQUIREMENTS,
      ...requirements,
      distalPosition:
        requirements.distalPosition ??
        (direction === "forward"
          ? editor.document.range.end
          : editor.document.range.start),
    };
```

Rather than copying the logic it was easier to just rewrite this class to use the base class, which Andreas says should have been done originally, just didn't think of it